ui-v2/app/templates/components/discovery-chain.hbs

```diff
@@ -0,0 +1,126 @@
+      </span>
+    </h2>
+  </header>
+  <div role="group">
+    {{#each (sort-by 'Name' splitters) as |item|}}
+      {{splitter-card item=item onclick=(action 'click')}}
+    {{/each}}
+  </div>
+</div>
+<div class="resolvers">
+  <header>
+    <h2>
+      Resolvers
+      <span>
+        <em role="tooltip">Resolvers are used to define which instances of a service should satisfy discovery requests.</em>
+      </span>
+    </h2>
+  </header>
+  <div role="group">
+    {{#each (sort-by 'Name' resolvers) as |item|}}
+      {{resolver-card item=item onclick=(action 'click')}}
+    {{/each}}
+  </div>
+</div>
+<svg width="100%" height="100%" viewBox={{concat '0 0 ' width ' ' height}} preserveAspectRatio="none">
+  {{#each routes as |item|}}
+    {{#let (dom-position (concat '#' item.ID)) as |src|}}
+      {{#let (dom-position (concat '#' item.NextNode)) as |dest|}}
+      <path
+        id={{concat item.ID '>' item.NextNode}}
+        d={{
+          svg-curve (hash
+            x=dest.x
+            y=(add dest.y (div dest.height 2))
+          ) src=(hash
+            x=(add src.x src.width)
+            y=(add src.y (div src.height 2))
+      )}} />
+      {{/let}}
+    {{/let}}
+  {{/each}}
+  {{#each splitters as |splitter|}}
+    {{#let (dom-position (concat '#' splitter.ID)) as |src|}}
+      {{#each splitter.Splits as |item|}}
+        {{#let (dom-position (concat '#' item.NextNode)) as |dest|}}
+    <path
+      id={{concat 'splitter:' splitter.Name '>' item.NextNode}}
+      class="split"
+      data-percentage={{item.Weight}}
+      d={{
+        svg-curve (hash
+          x=dest.x
+          y=(add dest.y (div dest.height 2))
+        ) src=(hash
+          x=(add src.x src.width)
+          y=(add src.y (div src.height 2))
+      )}} />
+        {{/let}}
+      {{/each}}
+    {{/let}}
+  {{/each}}
+</svg>
+<svg class="resolver-inlets" viewBox={{concat '0 0 10 ' height}}>
+  {{#each routes as |item|}}
+    {{#if (starts-with 'resolver:' item.NextNode) }}
+      {{#let (dom-position (concat '#' item.NextNode)) as |dest|}}
+        <circle r="3.5" cx="5" cy={{add dest.y (div dest.height 2)}} />
+      {{/let}}
+    {{/if}}
+  {{/each}}
+  {{#each splitters as |item|}}
+      {{#each item.Splits as |item|}}
+        {{#let (dom-position (concat '#' item.NextNode)) as |dest|}}
+          <circle r="3.5" cx="5" cy={{add dest.y (div dest.height 2)}} />
+        {{/let}}
+      {{/each}}
+  {{/each}}
+</svg>
+<svg class="splitter-inlets" viewBox={{concat '0 0 10 ' height}}>
+  {{#each routes as |item|}}
+    {{#if (starts-with 'splitter:' item.NextNode) }}
+      {{#let (dom-position (concat '#' item.NextNode)) as |dest|}}
+        <circle r="3.5" cx="5" cy={{add dest.y (div dest.height 2)}} />
+      {{/let}}
+    {{/if}}
+  {{/each}}
+</svg>
+<div class={{concat 'tooltip' (if activeTooltip ' active' '')}} style={{{ concat 'position: absolute;top:' y 'px;left:' x 'px;'}}}>
+  <span role="tooltip">{{round tooltip decimals=2}}%</span>
+</div>
```

Comment:
I appreciate that all the DOM lives in templates instead of dynamically added in `didReceiveAttrs`.